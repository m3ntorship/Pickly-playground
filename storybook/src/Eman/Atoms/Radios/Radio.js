import React, { useState } from "react";
import PropTypes from "prop-types";

export const RADIO = {
  SIZE: {
    DEFAULT: "Default",
    SMALL: "Small",
  },
};

export const Radio = ({ size, disabled, id }) => {
  const [isChecked, setIsChecked] = useState(false);
  const [value, setValue] = useState("");

  const handleCheck = () => {
    setIsChecked(!isChecked);
  };
  const handleChange = (e) => {
    setValue(e.target.id);
  };

  const divSize = size === RADIO.SIZE.DEFAULT ? "w-6 h-6" : "w-4 h-4";
  const spanSize = size === RADIO.SIZE.DEFAULT ? "w-2.5 h-2.5" : "w-1.5 h-1.5";
  return (
    <>
      {isChecked && (
        <div
          className={`relative rounded-lg bg-primary flex justify-center items-center hover:bg-primary-shd8 focus:ring-2 focus:ring-offset-primary-shd6 focus:outline-none ${divSize} ${
            disabled && "bg-primary-shd7"
          }`}
          onClick={handleCheck}
        >
          <span className={`bg-white rounded-lg ${spanSize}`}></span>
          <input
            onChange={handleChange}
            disabled={disabled}
            value={value}
            id={id}
            name="tab"
            type="radio"
            className="opacity-0 absolute cursor-pointer w-full h-full"
          />
        </div>
      )}
      {!isChecked && (
        <div
          className={`relative border border-grey-shd4 rounded-lg flex justify-center items-center hover:border-grey-shd1 focus:ring-2 focus:border-offset-grey-shd2 focus:outline-none ${divSize} ${
            disabled && "border-2 border-grey-shd6"
          }`}
          onClick={handleCheck}
        >
          <input
            onChange={handleChange}
            disabled={disabled}
            value={value}
            id={id}
            name="tab"
            type="radio"
            className="opacity-0 absolute cursor-pointer w-full h-full"
          />
        </div>
      )}
    </>
  );
};

Radio.propTypes = {
  size: PropTypes.oneOf(["Default", "Small"]),
  disabled: PropTypes.bool,
  value: PropTypes.string,
};
