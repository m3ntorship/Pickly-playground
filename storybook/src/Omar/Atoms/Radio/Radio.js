import React, { useState } from "react";
import PropTypes from "prop-types";
export const RADIO = {
  SIZE: {
    DEFAULT: "default",
    SMALL: "small",
  },
};
export const Radio = (props) => {
  const { disabled, size, label1, label2 } = props;
  const [radio1, setRadio1] = useState(true);
  const [radio2, setRadio2] = useState(false);
  const handleChange1 = () => {
    setRadio1(true);
    setRadio2(false);
  };
  const handleChange2 = () => {
    setRadio2(true);
    setRadio1(false);
  };

  const radioStyle1 = radio1 ? "bg-primary hover:bg-primary-shd8" : "";
  const radioStyle2 = radio2 ? "bg-primary hover:bg-primary-shd8" : "";
  return (
    <>
      <label htmlFor="radio-label1" className={"cursor-pointer"}>
        {label1}
        <div
          className={` ${radioStyle1} relative rounded-lg  border border-grey-shd4 flex justify-center items-center focus:ring-2 focus:ring-offset-primary-shd6 focus:outline-none ${
            size === RADIO.SIZE.SMALL ? "w-4 h-4" : "w-6 h-6"
          } ${disabled && "border-2 border-grey-shd6 bg-white hover:bg-white"}`}
        >
          <input
            data-testid="radio1"
            type="radio"
            className={`opacity-0 absolute cursor-pointer ${
              disabled && "cursor-not-allowed"
            }`}
            onChange={handleChange1}
            id="radio-label1"
            name="radio"
            disabled={disabled}
            checked={radio1}
          />
          <div
            className={`${
              size === RADIO.SIZE.SMALL ? "w-1.5 h-1.5" : "w-2.5 h-2.5"
            } rounded-lg bg-white`}
          ></div>
        </div>
      </label>
      <label htmlFor="radio-label2" className={"cursor-pointer"}>
        {label2}
        <div
          className={`${radioStyle2} relative border  border-grey-shd4 rounded-lg flex justify-center items-center hover:border-grey-shd1 focus:ring-2 focus:border-offset-grey-shd2 focus:outline-none ${
            size === RADIO.SIZE.SMALL ? "w-4 h-4" : "w-6 h-6"
          } ${disabled && "border-2 border-grey-shd6"}`}
        >
          <input
            data-testid="radio2"
            type="radio"
            className={`opacity-0 absolute  cursor-pointer w-full h-full ${
              disabled && "cursor-not-allowed"
            }`}
            onChange={handleChange2}
            id="radio-label2"
            name="radio"
            disabled={disabled}
            checked={radio2}
          />
          <div
            className={`${
              size === RADIO.SIZE.SMALL ? "w-1.5 h-1.5" : "w-2.5 h-2.5"
            } rounded-lg bg-white`}
          ></div>
        </div>
      </label>
    </>
  );
};

Radio.propTypes = {
  disabled: PropTypes.bool,
  size: PropTypes.oneOf(["default", "small"]),
  label1: PropTypes.string,
  label2: PropTypes.string,
};
