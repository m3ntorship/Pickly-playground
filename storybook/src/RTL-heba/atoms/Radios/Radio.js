import React from "react";
import PropTypes from "prop-types";

export const RADIO = {
  STATUS: {
    ON: "on",
    OFF: "off",
  },
  SIZE: {
    DEFAULT: "Default",
    SMALL: "Small",
  },
};

export const Radio = ({ status, size }) => {
  return (
    <>
      {size === RADIO.SIZE.DEFAULT && status === RADIO.STATUS.ON && (
        <button className="w-6 h-6 rounded-lg bg-primary flex justify-center items-center hover:bg-primary-shd8 focus:border-2 focus:border-primary-shd6 focus:outline-none">
          <span className="w-2.5 h-2.5 bg-white rounded-lg"></span>
        </button>
      )}
      {size === RADIO.SIZE.DEFAULT && status === RADIO.STATUS.OFF && (
        <button className="w-6 h-6 border border-grey-shd4 rounded-lg flex justify-center items-center hover:border-grey-shd1 focus:border-2 focus:border-grey-shd2 focus:outline-none"></button>
      )}
      {size === RADIO.SIZE.SMALL && status === RADIO.STATUS.ON && (
        <button className="w-4 h-4 rounded-lg bg-primary flex justify-center items-center hover:bg-primary-shd8 focus:border-2 focus:border-primary-shd6 focus:outline-none">
          <span className="w-1.5 h-1.5 bg-white rounded-lg"></span>
        </button>
      )}
      {size === RADIO.SIZE.SMALL && status === RADIO.STATUS.OFF && (
        <button className="w-4 h-4 border border-grey-shd4 rounded-lg flex justify-center items-center hover:border-grey-shd1 focus:border-2 focus:border-grey-shd2 focus:outline-none"></button>
      )}
    </>
  );
};

Radio.propTypes = {
  status: PropTypes.oneOf(["on", "off"]),
  size: PropTypes.oneOf(["Default", "Small"]),
};
