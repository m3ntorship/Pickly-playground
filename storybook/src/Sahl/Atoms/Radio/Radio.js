import React from "react";
import PropTypes from "prop-types";

export default function Radio(props) {
  const {
    value,
    name,
    disabled = false,
    label,
    size = "default",
    className,
    onChange,
    checked = "false",
  } = props;

  //////////////////////////////////
  // Classes
  ///////////
  // label Classes
  let labelClasses = [
    "flex",
    "items-center",
    "justify-end",
    "relative",
    "cursor-pointer",
    "rounded-full",
    "group",
  ];
  // Add the props classes
  if (className)
    labelClasses = [...labelClasses, className];

  // CustomRadio Classes
  let customRadioClasses = [
    "absolute",
    "inset-0",
    "m-auto",
    "border",
    "border-solid",
    "rounded-full",
  ];
  // Dot Classes
  const dotClasses = [
    "bg-white",
    "rounded-full",
    "w-1/2",
    "h-1/2",
    "absolute",
    "z-10",
    "inset-0",
    "m-auto",
  ];
  // Change colors if the raido checked
  if (checked) {
    customRadioClasses = [
      ...customRadioClasses,
      "border-primary",
      "bg-primary",
      "group-hover:bg-primary-shd8",
    ];
    dotClasses.push("block");
  } else {
    customRadioClasses = [
      ...customRadioClasses,
      "border-grey-shd4",
      "bg-white",
      "group-hover:border-grey",
    ];
    dotClasses.push("hidden");
  }

  // Change size
  if (size === "default") {
    customRadioClasses = [...customRadioClasses, "h-4", "w-4"];
    if (label) dotClasses.push("ml-12");
  } else {
    customRadioClasses = [...customRadioClasses, "h-6", "w-6"];
    if (label) dotClasses.push("ml-14");
  }

  // Change opacity if disabled
  if (disabled) labelClasses.push("opacity-40");

  return (
    <label className={`${labelClasses.join(" ")}`}>
      <input
        disabled={disabled}
        type="radio"
        onChange={onChange}
        checked={checked}
        className="opacity-0 relative z-10"
        name={name}
        value={value}
      />
      {/* CustomRadio */}
      <div className={`${customRadioClasses.join(" ")}`}>
        {/* Dot */}
        <div className={`${dotClasses.join(" ")}`}></div>
      </div>
      <span className={`block`}>{label}</span>
    </label>
  );
}

Radio.propTypes = {
  size: PropTypes.oneOf(["default", "large"]),
  label: PropTypes.string,
  disabled: PropTypes.bool,
  className: PropTypes.string,
  name: PropTypes.string.isRequired,
};
