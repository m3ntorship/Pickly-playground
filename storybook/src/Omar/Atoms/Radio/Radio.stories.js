import React from "react";
import { Radio, RADIO } from "./Radio";

export default {
  title: "Omar/Atoms/Radio",
  component: Radio,
  argsTypes: {
    size: {
      control: {
        type: "select",
        options: [RADIO.SIZE.DEFAULT, RADIO.SIZE.SMALL],
      },
    },
  },
};

const Template = (args) => <Radio {...args} />;
export const Default = Template.bind({});
Default.args = {
  size: RADIO.SIZE.SMALL,
  disabled: false,
};
