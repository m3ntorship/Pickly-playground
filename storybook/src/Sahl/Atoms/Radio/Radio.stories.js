import React from "react";

import Radio from "./Radio";

const defaultExport = {
  title: "Sahl/Atoms/Radio",
  component: Radio,
};

export default defaultExport;

const Template = (args) => <Radio {...args} />;

export const Default = Template.bind({});
Default.args = {
  size: "default",
  disabled: false,
  label: "test",
};
