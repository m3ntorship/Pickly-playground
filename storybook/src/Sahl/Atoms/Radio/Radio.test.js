import React from "react";
import renderer from "react-test-renderer";
import Radio from "./Radio";

describe("Radio Component", () => {
  it("should give a console error when running without any props", () => {
    const tree = renderer.create(<Radio />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render correctly when passing only name prop", () => {
    const tree = renderer.create(<Radio name="polls" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render with default size without passing size prop", () => {
    const tree = renderer.create(<Radio name="polls" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render with large size when size='large'", () => {
    const tree = renderer.create(<Radio name="polls" size='large'/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render with disabled when passing disabled", () => {
    const tree = renderer.create(<Radio name="polls" disabled/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render without label by default", () => {
    const tree = renderer.create(<Radio name="polls" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render with label when passing label='testLabel'", () => {
    const tree = renderer.create(<Radio name="polls" label='testLabel'/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
