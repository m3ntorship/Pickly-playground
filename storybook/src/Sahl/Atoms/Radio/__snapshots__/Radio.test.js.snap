// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio Component should give a console error when running without any props 1`] = `
<label
  className="flex items-center justify-end relative cursor-pointer rounded-full group"
>
  <input
    checked="false"
    className="opacity-0 relative z-10"
    disabled={false}
    type="radio"
  />
  <div
    className="absolute inset-0 m-auto border border-solid rounded-full border-primary bg-primary group-hover:bg-primary-shd8 h-4 w-4"
  >
    <div
      className="bg-white rounded-full w-1/2 h-1/2 absolute z-10 inset-0 m-auto block"
    />
  </div>
  <span
    className="block"
  />
</label>
`;

exports[`Radio Component should render correctly when passing only name prop 1`] = `
<label
  className="flex items-center justify-end relative cursor-pointer rounded-full group"
>
  <input
    checked="false"
    className="opacity-0 relative z-10"
    disabled={false}
    name="polls"
    type="radio"
  />
  <div
    className="absolute inset-0 m-auto border border-solid rounded-full border-primary bg-primary group-hover:bg-primary-shd8 h-4 w-4"
  >
    <div
      className="bg-white rounded-full w-1/2 h-1/2 absolute z-10 inset-0 m-auto block"
    />
  </div>
  <span
    className="block"
  />
</label>
`;

exports[`Radio Component should render with default size without passing size prop 1`] = `
<label
  className="flex items-center justify-end relative cursor-pointer rounded-full group"
>
  <input
    checked="false"
    className="opacity-0 relative z-10"
    disabled={false}
    name="polls"
    type="radio"
  />
  <div
    className="absolute inset-0 m-auto border border-solid rounded-full border-primary bg-primary group-hover:bg-primary-shd8 h-4 w-4"
  >
    <div
      className="bg-white rounded-full w-1/2 h-1/2 absolute z-10 inset-0 m-auto block"
    />
  </div>
  <span
    className="block"
  />
</label>
`;

exports[`Radio Component should render with disabled when passing disabled 1`] = `
<label
  className="flex items-center justify-end relative cursor-pointer rounded-full group opacity-40"
>
  <input
    checked="false"
    className="opacity-0 relative z-10"
    disabled={true}
    name="polls"
    type="radio"
  />
  <div
    className="absolute inset-0 m-auto border border-solid rounded-full border-primary bg-primary group-hover:bg-primary-shd8 h-4 w-4"
  >
    <div
      className="bg-white rounded-full w-1/2 h-1/2 absolute z-10 inset-0 m-auto block"
    />
  </div>
  <span
    className="block"
  />
</label>
`;

exports[`Radio Component should render with label when passing label='testLabel' 1`] = `
<label
  className="flex items-center justify-end relative cursor-pointer rounded-full group"
>
  <input
    checked="false"
    className="opacity-0 relative z-10"
    disabled={false}
    name="polls"
    type="radio"
  />
  <div
    className="absolute inset-0 m-auto border border-solid rounded-full border-primary bg-primary group-hover:bg-primary-shd8 h-4 w-4"
  >
    <div
      className="bg-white rounded-full w-1/2 h-1/2 absolute z-10 inset-0 m-auto block ml-12"
    />
  </div>
  <span
    className="block"
  >
    testLabel
  </span>
</label>
`;

exports[`Radio Component should render with large size when size='large' 1`] = `
<label
  className="flex items-center justify-end relative cursor-pointer rounded-full group"
>
  <input
    checked="false"
    className="opacity-0 relative z-10"
    disabled={false}
    name="polls"
    type="radio"
  />
  <div
    className="absolute inset-0 m-auto border border-solid rounded-full border-primary bg-primary group-hover:bg-primary-shd8 h-6 w-6"
  >
    <div
      className="bg-white rounded-full w-1/2 h-1/2 absolute z-10 inset-0 m-auto block"
    />
  </div>
  <span
    className="block"
  />
</label>
`;

exports[`Radio Component should render without label by default 1`] = `
<label
  className="flex items-center justify-end relative cursor-pointer rounded-full group"
>
  <input
    checked="false"
    className="opacity-0 relative z-10"
    disabled={false}
    name="polls"
    type="radio"
  />
  <div
    className="absolute inset-0 m-auto border border-solid rounded-full border-primary bg-primary group-hover:bg-primary-shd8 h-4 w-4"
  >
    <div
      className="bg-white rounded-full w-1/2 h-1/2 absolute z-10 inset-0 m-auto block"
    />
  </div>
  <span
    className="block"
  />
</label>
`;
